### PR TITLE
More lgccps changes to make it work

### PR DESCRIPTION
**Pull request by ruiling**
This is the set of changes to make lgccps lowering work correctly for continufy based ray-tracing.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 6fdd4219e8590602d6fd4c527a61acd17a3cb41a
<details><summary>CTS tests (Failed: 0/208497)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69503 (53.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 32525/69503 (46.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28360/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/PatchEntryPointMutate.cpp [hunk 1]
```diff
@@ -502,8 +516,14 @@ Function *PatchEntryPointMutate::lowerCpsFunction(Function *func, ArrayRef<Type
 
   AttributeList oldAttrs = func->getAttributes();
   SmallVector<AttributeSet, 8> argAttrs;
-  for (unsigned idx = 0; idx != userDataTys.size(); ++idx)
+  bool hasLocalInvocationIdArg = argNames.back() == "LocalInvocationId";
```
Comment (Flakebi) [inline lgc/patch/PatchEntryPointMutate.cpp:524]: Depending on the name of an argument seems rather brittle. Is there any better way how we can find out if we have a local invocation id (like the condition for `cpsShaderInputs` below)?

File: lgc/patch/PatchEntryPointMutate.cpp [hunk 2]
```diff
@@ -408,12 +410,15 @@ bool PatchEntryPointMutate::lowerCpsOps(Function *func, ShaderInputs *shaderInpu
...
 
   if (isCpsFunc) {
-    // FIXME: Switch to LLVM intrinsic
-    FunctionType *setInactiveChainTy = FunctionType::get(vcrTy, {vcrTy, vcrTy}, false);
-    auto setInactiveChain = Function::Create(setInactiveChainTy, GlobalValue::ExternalLinkage,
-                                             "llvm.amdgcn.setinactive.chain.arg", func->getParent());
     auto *vcrShaderArg = func->getArg(numShaderArg);
-    vcr = builder.CreateCall(setInactiveChain, {vcr, vcrShaderArg});
+    // When we are working with LLVM version without the llvm.amdgcn.set.inactive.chain.arg, we cannot simply declare
+    // it and call it. LLVM will mis-recognize as llvm.amdgcn.set.inactive, and lit-test would just fail. So here we
```
Comment (Flakebi) [inline lgc/patch/PatchEntryPointMutate.cpp:415]: typo: [misrecognize](https://en.wiktionary.org/wiki/misrecognize)?

**Comment by amdvlk-admin** [comment]
### Test summary for commit 6fdd4219e8590602d6fd4c527a61acd17a3cb41a
<details><summary>CTS tests (Failed: 0/138994)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28360/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit f9e42bcc1114d85ce0ff480d684e1412a0de7f68
<details><summary>CTS tests (Failed: 0/138994)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28360/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/PatchEntryPointMutate.cpp (hunk 1 above)
Comment (ruiling) [reply lgc/patch/PatchEntryPointMutate.cpp]: I just add one boolean argument for it.

**Review by Flakebi — APPROVED** [review]

**Pull request merged**